app/lib/actions/remote_execution/run_hosts_job.rb

```diff
@@ -71,6 +77,16 @@ def determine_proxy(template_invocation, host, load_balancer)
 
         proxy
       end
+
+      def provider_job_class(provider)
+        case provider
+        when 'SSH'
+          RunSSHJob
```

Comment:
What about putting it `here https://github.com/theforeman/foreman_remote_execution/blob/v0.3.0/app/models/remote_execution_provider.rb#L22` to make it provider-specific?